corehq/motech/repeaters/tasks.py

```diff
@@ -66,15 +69,36 @@ def clean_logs():
     queue=settings.CELERY_PERIODIC_QUEUE,
 )
 def check_repeaters():
+    # this creates a task for all partitions
+    # the Nth child task determines if a lock is available for the Nth partition
+    for current_partition in range(CHECK_REPEATERS_PARTITION_COUNT):
+        check_repeaters_in_partition.delay(current_partition, CHECK_REPEATERS_PARTITION_COUNT)
+
+
+def iterate_repeat_record_ids_for_partition(start, partition, total_partitions):
+    for record_id in iterate_repeat_record_ids(start, chunk_size=100000):
+        if hash(record_id) % total_partitions == partition:
+            yield record_id
+
+
+def iterate_repeat_records_for_partition(start, partition, total_partitions):
+    # chunk the fetching of documents from couch
+    for chunked_ids in chunked(iterate_repeat_record_ids_for_partition(start, partition, total_partitions), 10000):
+        yield from iterate_repeat_records_for_ids(chunked_ids)
```

Comment:
What is the pythonic way of defining these methods? I prefer to keep methods concise which is why I pulled these out, but I will admit it feels wrong to have methods defined amongst `@task` decorated methods in a tasks.py file. A nested definition is ok, but requires an unnecessary changing of parameters making it less readable. Should these methods live in a different file?

Reply:
I'd leave 'em in here, but prefix them with underscores so it's clear they're just little helpers.  If a single task ended up needing a _lot_ of backing code, I'd pull out the logical bulk of the task to a standalone file.